### PR TITLE
Run the pands udf using cudf on Databricks

### LINKED ISSUE #2026: [TEST] run the pands udf using cudf on Databricks
**Is your feature request related to a problem? Please describe.**
Right now we don't run the pandas udf using cudf tests on Databricks because we haven't setup the conda environment.

We should test this at some point at least manually during release process but ideally running nightly.

We have documentation on how to make that work here: https://confluence.nvidia.com/display/SPARK/How+to+enable+cuDF+Pandas+UDF+on+Azure+Databricks

If there is anything special the end user would need to do we should put in our regularly released documents.

### PR DESCRIPTION
**Pull request by NvTimLiu**
Fixes #2026 

Add the init script to set up environment for the cudf_udf tests on Databrcks, run cudf-udf tests nightly

Signed-off-by: Tim Liu <timl@nvidia.com>

### CODE REVIEW

**Comment by NvTimLiu** [comment]
build

**Comment by NvTimLiu** [comment]
Test PASS on the Databricks nightly pipeline and integration pipelines

File: jenkins/databricks/init_cudf_udf.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+# The initscript to set up environment for the cudf_udf tests on Databrcks
+# Will be automatically pushed into the dbfs:/databricks/init_scripts once it is updated.
+
+CUDF_VER=${CUDF_VER:-0.19}
+
+# Use mamba to install cudf-udf packages to speed up conda resolve time
+base=$(conda info --base)
+conda create -y -n mamba -c conda-forge mamba
+pip uninstall -y pyarrow
```
Comment (tgravescs) [inline jenkins/databricks/init_cudf_udf.sh:26]: the version of yarrow is to old or why uninstall here?

File: jenkins/databricks/test.sh [hunk 2]
```diff
@@ -33,10 +33,25 @@ sudo chmod 777 /databricks/data/logs/
 sudo chmod 777 /databricks/data/logs/*
 echo { \"port\":\"15002\" } > ~/.databricks-connect
 
+SPARK_SUBMIT_FLAGS="--conf spark.python.daemon.module=rapids.daemon_databricks \
+    --conf spark.rapids.memory.gpu.allocFraction=0.1 \
```
Comment (tgravescs) [inline jenkins/databricks/test.sh:37]: I assume this setting still works for the rest of the tests, did we see a change in runtime at all?

Comment (tgravescs) [reply jenkins/databricks/test.sh:37]: oh never mind, I see below we run them separate

Comment (tgravescs) [reply jenkins/databricks/test.sh:37]: it might be nice to rename these to be CUDF_UDF_TEST_ARGS

File: jenkins/databricks/test.sh [hunk 3]
```diff
@@ -33,10 +33,25 @@ sudo chmod 777 /databricks/data/logs/
...
+    --conf spark.rapids.python.memory.gpu.allocFraction=0.1 \
+    --conf spark.rapids.python.concurrentPythonWorkers=2"
+
 if [ -d "$LOCAL_JAR_PATH" ]; then
```
Comment (jlowe) [inline jenkins/databricks/test.sh:41]: Seems like we could reduce the redundancy of the two code paths here with something like this:
```bash
export LOCAL_JAR_PATH=${LOCAL_JAR_PATH:-/home/ubuntu/spark-rapids}
bash $LOCAL_JAR_PATH/integration_tests/run_pyspark_from_build.sh  --runtime_env="databricks"

## Run cudf-udf tests
PYTHON_JARS=$(ls $LOCAL_JAR_PATH/rapids-4-spark_*.jar | grep -v tests.jar)
SPARK_SUBMIT_FLAGS="$SPARK_SUBMIT_FLAGS --conf spark.executorEnv.PYTHONPATH="$PYTHON_JARS"
SPARK_SUBMIT_FLAGS=$SPARK_SUBMIT_FLAGS TEST_PARALLEL=1 \
        bash $LOCAL_JAR_PATH/integration_tests/run_pyspark_from_build.sh --runtime_env="databricks" -m "cudf_udf" --cudf_udf
```

File: jenkins/databricks/test.sh (hunk 2 above)
Comment (NvTimLiu) [reply jenkins/databricks/test.sh:37]: Good suggestion, let me change it

File: jenkins/databricks/test.sh (hunk 3 above)
Comment (NvTimLiu) [reply jenkins/databricks/test.sh:41]: Good suggestion, let me change it

File: jenkins/databricks/init_cudf_udf.sh (hunk 1 above)
Comment (NvTimLiu) [reply jenkins/databricks/init_cudf_udf.sh:26]: In fact the pyarrow version is the same (1.0.1).  The init script is from Zhu hao's confluence page, and I was told that if we do not uninstall pyarrow first, there will be some dependencies issues when running install/unstall "{base}/envs/mamba/bin/mamba remove -y c-ares zstd libprotobuf pandas", 

I tried to remove 'pip uninstall -y pyarrow' here and got python-jvm socket connection error as below, there should be some error like Socket parameters mismatching.

Caused by: java.net.SocketException: Connection reset
	at java.net.SocketInputStream.read(SocketInputStream.java:210)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at java.net.SocketInputStream.read(SocketInputStream.java:224)
	at java.io.DataInputStream.readInt(DataInputStream.java:387)

File: jenkins/databricks/test.sh (hunk 3 above)
Comment (jlowe) [reply jenkins/databricks/test.sh:41]: If it's complicated to commonize this code, that's fine let's leave that suggestion out of this PR.  We can do a followup PR to tackle it.

**Review by jlowe — APPROVED** [review]

**Comment by NvTimLiu** [comment]
build

**Pull request merged**